File: Tyche.Android/session/src/main/java/com/felipearpa/tyche/session/authentication/application/SendSignInLinkToEmailUseCase.kt
package com.felipearpa.tyche.session.authentication.application

import com.felipearpa.tyche.core.type.Email
import com.felipearpa.tyche.session.authentication.domain.AuthenticationRepository

class SendSignInLinkToEmailUseCase(private val authenticationRepository: AuthenticationRepository) {
    suspend fun execute(email: Email) =
        authenticationRepository.sendSignInLinkToEmail(email = email.value)
}

File: iOS/Session/Sources/Session/Authentication/Application/SendSignInLinkToEmailUseCase.swift
import Core

public class SendSignInLinkToEmailUseCase {
    private let authenticationRepository: AuthenticationRepository
    
    init(authenticationRepository: AuthenticationRepository) {
        self.authenticationRepository = authenticationRepository
    }
    
    public func execute(email: Email) async -> Result<Void, Error> {
        await authenticationRepository.sendSignInLinkToEmail(email: email.value)
    }
}
